claimManagement/src/main/java/org/openimis/imisclaims/domain/entity/PendingClaim.java

```diff
@@ -0,0 +1,423 @@
+    }
+
+    @Nullable
+    public String getIcdCode4() {
+        return icdCode4;
+    }
+
+    @NonNull
+    public String getVisitType() {
+        return visitType;
+    }
+
+    @NonNull
+    public List<Service> getServices() {
+        return services;
+    }
+
+    @NonNull
+    public List<Medication> getMedications() {
+        return medications;
+    }
+
+    public static final Creator<PendingClaim> CREATOR = new Creator<>() {
+        @Override
+        public PendingClaim createFromParcel(Parcel in) {
+            return new PendingClaim(in);
+        }
+
+        @Override
+        public PendingClaim[] newArray(int size) {
+            return new PendingClaim[size];
+        }
+    };
+
+    public static class Service implements Parcelable {
+        @NonNull
+        public static List<Service> fromJson(@NonNull JSONArray array) throws Exception {
+            List<Service> list = new ArrayList<>();
+            for (int i = 0; i < array.length(); i++) {
+                JSONObject medication = array.getJSONObject(i);
+                list.add(new Service(
+                        /* code = */ medication.getString("ServiceCode"),
+                        /* price = */ Double.parseDouble(medication.getString("ServicePrice")),
+                        /* quantity = */ Integer.parseInt(medication.getString("ServiceQuantity"))
+                ));
+            }
+            return list;
+        }
+
+        @NonNull
+        private final String code;
+        private final double price;
+        private final double quantity;
+
+        public Service(
+                @NonNull String code,
+                double price,
+                double quantity
+        ) {
+            this.code = code;
+            this.price = price;
+            this.quantity = quantity;
+        }
+
+        protected Service(Parcel in) {
+            code = in.readString();
+            price = in.readDouble();
+            quantity = in.readDouble();
+        }
+
+        @Override
+        public void writeToParcel(Parcel dest, int flags) {
+            dest.writeString(code);
+            dest.writeDouble(price);
+            dest.writeDouble(quantity);
+        }
+
+        @Override
+        public int describeContents() {
+            return 0;
+        }
+
+        @NonNull
+        public String getCode() {
+            return code;
+        }
+
+        public double getPrice() {
+            return price;
+        }
+
+        public double getQuantity() {
+            return quantity;
+        }
+
+        public static final Creator<Service> CREATOR = new Creator<>() {
+            @Override
+            public Service createFromParcel(Parcel in) {
+                return new Service(in);
+            }
+
+            @Override
+            public Service[] newArray(int size) {
+                return new Service[size];
+            }
+        };
+    }
+
+    public static class Medication implements Parcelable {
```

Comment:
why not using the existing class? they look alike.

Reply:
It has different properties